### ISSUE TITLE
Not receiving an R2

### ISSUE THREAD

**Issue by sweetmoniker** [franzmohr/prais#3]
First of all, thanks for this package. I'm looking at the following data and getting the following result.

```
dep = c(59512,69565,78395,86085,68288,76692,86647,80146,76118,79969,82338,79269,68427,80309,87515,81462,62390,67105)

ind = c(0.12820512821,0.20512820513,0.16666666667,0.11904761905,0.0625,0.08333333333,0.11494252874,0.17857142857,0.16049382716,0.13636363636,0.16842105263,0.136842105263158,0.161616161616162,0.105882352941176,0.150442477876106,0.176470588235294,0.101123595505618,0.138888888888889)

summary(prais_winsten(dep ~ ind, data = data.frame(dep, ind)))
Iteration 0: rho = 0
Iteration 1: rho = 0.2433
Iteration 2: rho = 0.2662
Iteration 3: rho = 0.2688
Iteration 4: rho = 0.2691
Iteration 5: rho = 0.2691
Iteration 6: rho = 0.2691
Iteration 7: rho = 0.2691

Call:
prais_winsten(formula = dep ~ ind, data = data.frame(dep, ind))

Residuals:
   Min     1Q Median     3Q    Max 
-15618  -7152   3157   5068  12154 

AR(1) coefficient rho after 7 Iterations: 0.2691

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)    68970       8124   8.490 2.54e-07 ***
ind            48051      55657   0.863    0.401    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 8158 on 16 degrees of freedom

Durbin-Watson statistic (original): 1.248 
Durbin-Watson statistic (transformed): 1.556
```
I'm not sure why it's not outputting an r2. If I save the regression as a variable and access the r.squared value it returns 'NULL.' The Durbin Watson test returns the following DW = 1.248, p-value = 0.04158. Does the independent variable need to be significant to return a r2? Thanks!

**Comment by franzmohr** [comment]
Many thanks for your appreciation! Love to hear that people are using the package.

Going through the functions step-by-step, I noticed in `summary.prais` that the sum of squared residuals of your model specification is larger than the total sum of squares, which leads to negative R-squared values. Therefore, the R-squared is not calculated. I also just ran your example in a former version of Stata, which also refuses to print an R-squared.

Maybe there is an issue with the scale. Trying log(dep) produces:

```
summary(prais_winsten(log(dep) ~ ind, data = data.frame(dep, ind)))
Iteration 0: rho = 0
Iteration 1: rho = 0.2542
Iteration 2: rho = 0.2811
Iteration 3: rho = 0.2846
Iteration 4: rho = 0.2851
Iteration 5: rho = 0.2851
Iteration 6: rho = 0.2851
Iteration 7: rho = 0.2851
Iteration 8: rho = 0.2851

Call:
prais_winsten(formula = log(dep) ~ ind, data = data.frame(dep, 
    ind))

Residuals:
     Min       1Q   Median       3Q      Max 
-0.22535 -0.09241  0.04539  0.06910  0.15975 

AR(1) coefficient rho after 8 Iterations: 0.2851

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  11.1281     0.1101 101.109   <2e-16 ***
ind           0.7113     0.7525   0.945    0.359    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.1104 on 16 degrees of freedom
Multiple R-squared:  0.9677,	Adjusted R-squared:  0.9656 
F-statistic: 478.7 on 1 and 16 DF,  p-value: 2.384e-13

Durbin-Watson statistic (original):   1.2 
Durbin-Watson statistic (transformed): 1.542
```

This is a suspiciously high R-squared value. 

I also tried a differenced version that seems quite promising:

```
data <- data.frame(dep = diff(log(dep)), ind = ind[-1])
  
summary(prais_winsten(dep ~ ind, data = data))

Iteration 0: rho = 0
Iteration 1: rho = -0.0964
Iteration 2: rho = -0.0969
Iteration 3: rho = -0.0969
Iteration 4: rho = -0.0969

Call:
prais_winsten(formula = dep ~ ind, data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.24529 -0.10487  0.04524  0.10202  0.17811 

AR(1) coefficient rho after 4 Iterations: -0.09693

Coefficients:
            Estimate Std. Error t value Pr(>|t|)
(Intercept) -0.09478    0.12278  -0.772    0.452
ind          0.72519    0.85825   0.845    0.411

Residual standard error: 0.1326 on 15 degrees of freedom
Multiple R-squared:  0.04478,	Adjusted R-squared:  -0.0189 
F-statistic: 0.7031 on 1 and 15 DF,  p-value: 0.4149

Durbin-Watson statistic (original): 2.135 
Durbin-Watson statistic (transformed): 2.038
```

Though it might suggest that you don't need the Prais-Winsten-estimator, since the dependent variable was rendered stationary by taking the first difference.

**Comment by sweetmoniker** [comment]
Very interesting. I was wondering if the R2 might be negative. I realize that the number of observations is less than ideal for time series, but I've got what I've got... The scale is an interesting point that hadn't occurred to me. I built some code that parses prais results and not having an r2 broke it. Half of me suspected that this was "by design." But now if I need an r2 to report it looks like I have some options. Thanks for looking into it. Cheers.
